src/tests/ftest/nvme/nvme_health.py

```diff
@@ -0,0 +1,127 @@
+        self.start_servers()
+
+        single_pool_nvme_size = int((storage * 0.80)/no_of_pools)
+        pools = []
+        #Create the Large number of pools
+        for _pool in range(no_of_pools):
+            pool = TestPool(self.context, dmg_command=self.get_dmg_command())
+            pool.get_params(self)
+            #SCM size is 10% of NVMe
+            pool.scm_size.update('{}GB'.format(single_pool_nvme_size * 0.10))
+            pool.nvme_size.update('{}GB'.format(single_pool_nvme_size))
+            pool.create()
+            pools.append(pool)
+
+        #initialize the dmg command
+        self.dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        self.dmg.get_params(self)
+        self.dmg.insecure.update(
+            self.server_managers[0].get_config_value("allow_insecure"),
+            "dmg.insecure")
+
+        #Get Pool query for SMD
+        self.dmg.set_sub_command("storage")
+        self.dmg.sub_command_class.set_sub_command("query")
+        self.dmg.sub_command_class.sub_command_class.set_sub_command("smd")
+        for host in self.hostlist_servers:
+            self.dmg.sub_command_class. \
+                sub_command_class.sub_command_class.devices.value = False
+            self.dmg.sub_command_class. \
+                sub_command_class.sub_command_class.pools.value = True
+            self.dmg.hostlist = host
+            try:
+                result = self.dmg.run()
+            except CommandFailure as error:
+                self.fail("dmg command failed: {}".format(error))
+            #Verify all pools UUID listed as part of query
+            for pool in pools:
+                if pool.uuid.lower() not in result.stdout:
+                    self.fail('Pool uuid {} not found in smd query'
+                              .format(pool.uuid.lower()))
+
+        # Get the device ID
+        device_ids = get_device_ids(self.dmg, self.hostlist_servers)
+
+        #Get the device states
+        for host in device_ids:
+            self.dmg.hostlist = host
+            for _dev in device_ids[host]:
+                try:
+                    result = self.dmg.storage_query_device_state(_dev)
+                except CommandFailure as details:
+                    self.fail("dmg get device states failed {}".format(details))
+                if 'State: NORMAL' not in result.stdout:
+                    self.fail("device {} on host {} is not NORMAL"
+                              .format(_dev, host))
+
+        #Get the blobstore health data
+        for host in device_ids:
+            self.dmg.hostlist = host
+            for _dev in device_ids[host]:
+                try:
+                    self.dmg.storage_query_blobstore(_dev)
+                except CommandFailure as error:
+                    self.fail("dmg get blobstore health failed {}"
+                              .format(error))
+        #Check nvme-health command works
+        try:
+            self.dmg.storage_query_nvme_health()
```

Comment:
(pylint-no-member) Instance of 'DmgCommand' has no 'storage_query_nvme_health' member